sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThroughputControlGroup.java

```diff
@@ -0,0 +1,181 @@
+    ThroughputControlMode getControlMode() {
+        return this.controlMode;
+    }
+
+    /**
+     * Set throughput group control mode.
+     *
+     * By default it will be local control mode.
+     *
+     * @return the {@link ThroughputControlGroup}.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public ThroughputControlGroup setControlMode(ThroughputControlMode controlMode) {
+        checkNotNull(controlMode, "Throughput control mode cannot be null");
+        this.controlMode = controlMode;
+        return this;
+    }
+
+    /**
+     * Get the throughput control group name.
+     *
+     * @return the group name.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public String getGroupName() {
+        return this.groupName;
+    }
+
+    /**
+     * Get the throughput control group target container.
+     *
+     * @return the {@link CosmosAsyncContainer}.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public CosmosAsyncContainer getTargetContainer() {
+        return this.targetContainer;
+    }
+
+    /**
+     * Get throughput control group target throughput.
+     * Since we allow either TargetThroughput or TargetThroughputThreshold, this value can be null.
+     *
+     * By default, it is null.
+     *
+     * @return the target throughput.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public Integer getTargetThroughput() {
+        return this.targetThroughput;
+    }
+
+    /**
+     * Get the throughput control group target throughput threshold.
+     * Since we allow either TargetThroughput or TargetThroughputThreshold, this value can be null.
+     *
+     * By default, this value is null.
+     *
+     * @return the target throughput threshold.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public Double getTargetThroughputThreshold() {
+        return this.targetThroughputThreshold;
+    }
+
+    /**
+     * Get whether this throughput control group will be used by default.
+     *
+     * By default, it is false.
+     *
+     * @return {@code true} this throughput control group will be used by default unless being override. {@code false} otherwise.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public boolean isUseByDefault() {
+        return this.useByDefault;
+    }
+
+    /**
+     * Set the throughput control group to be used by default.
+     *
+     * @return the {@link ThroughputControlGroup}.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public ThroughputControlGroup setUseByDefault() {
```

Comment:
setUseByDefault -> setAsDefault

Reply:
removed setter, rename getter to isDefault